clients/secret/secret.go

```diff
@@ -0,0 +1,165 @@
+package secret
+
+import (
+	"context"
+	"fmt"
+	"github.com/influxdata/influx-cli/v2/api"
+	"github.com/influxdata/influx-cli/v2/clients"
+)
+
+type Client struct {
+	clients.CLI
+	api.SecretsApi
+	api.OrganizationsApi
+}
+
+type secretPrintOpt struct {
+	deleted bool
+	secret  secret
+	secrets []secret
+}
+
+type secret struct {
+	key   string
+	orgID string
+}
+
+type DeleteParams struct {
+	clients.OrgParams
+	Key string
+}
+
+func (c Client) Delete(ctx context.Context, params *DeleteParams) error {
+	orgID, err := c.getOrgID(ctx, params.OrgParams)
+	if err != nil {
+		return err
+	}
+
+	if err := c.PostOrgsIDSecrets(ctx, orgID).Execute(); err != nil {
+		return fmt.Errorf("failed to delete secret with key %q: %v", params.Key, err)
+	}
+
+	return c.printSecrets(secretPrintOpt{
+		deleted: true,
+		secret: secret{
+			key:   params.Key,
+			orgID: orgID,
+		},
+	})
+}
+
+type ListParams struct {
+	clients.OrgParams
+}
+
+func (c Client) List(ctx context.Context, params *ListParams) error {
+
+	orgID, err := c.getOrgID(ctx, params.OrgParams)
+	if err != nil {
+		return err
+	}
+
+	response, err := c.GetOrgsIDSecrets(ctx, orgID).Execute()
+	if err != nil {
+		return fmt.Errorf("failed to retrieve secret keys: %s", err)
+	}
+
+	secrets := response.GetSecrets()
+	opts := make([]secret, 0, len(secrets))
+	for _, entry := range secrets {
+		opts = append(opts, secret{
+			key:   entry,
+			orgID: orgID,
+		})
+	}
+	return c.printSecrets(secretPrintOpt{secrets: opts})
+}
+
+type UpdateParams struct {
+	clients.OrgParams
+	Key   string
+	Value string
+}
+
+func (c Client) Update(ctx context.Context, params *UpdateParams) error {
+	orgID, err := c.getOrgID(ctx, params.OrgParams)
+	if err != nil {
+		return err
+	}
+
+	var secretVal string
+	if params.Value != "" {
+		secretVal = params.Value
+	} else {
+		secretVal, err = c.StdIO.GetSecret("Please type your secret", 0)
+		if err != nil {
+			return err
+		}
+	}
+
+	err = c.PatchOrgsIDSecrets(ctx, orgID).RequestBody(map[string]string{
+		params.Key: secretVal,
+	}).Execute()
+	if err != nil {
+		return fmt.Errorf("failed to update secret with key %q: %v", params.Key, err)
+	}
+
+	return c.printSecrets(secretPrintOpt{
+		secret: secret{
+			key:   params.Key,
+			orgID: orgID,
+		},
+	})
+}
+
+func (c Client) printSecrets(opts secretPrintOpt) error {
+	if c.PrintAsJSON {
+		var v interface{} = opts.secrets
+		if opts.secrets == nil {
+			v = opts.secret
+		}
+		return c.PrintJSON(v)
+	}
+
+	headers := []string{"Key", "Organization ID"}
+	if opts.deleted {
+		headers = append(headers, "Deleted")
+	}
+	if opts.secrets == nil {
```

Comment:
It probably doesn't matter since we only use 1 or the other, but I think the pattern used in other commands is instead:
```suggestion
	if opts.secret != nil {
```

Reply:
I'm not totally understanding why, but when I attempt to make this change, my IDE is telling me that `nil cannot be reassigned to type "secret"`. This almost feels like an IDE error, though there is probably a good explanation

Reply:
Ah, it's because you've typed `opts.secret` as `secret` instead of `*secret`